src/util-thash.c

```diff
@@ -483,12 +560,16 @@ static inline int THashCompare(const THashConfig *cnf, void *a, void *b)
 static THashData *THashDataGetNew(THashTableContext *ctx, void *data)
 {
     THashData *h = NULL;
+    uint32_t len = 0;
+    if (ctx->config.DataSize) {
+        len = ctx->config.DataSize(data);
+    }
 
     /* get data from the spare queue */
     h = THashDataDequeue(&ctx->spare_q);
     if (h == NULL) {
         /* If we reached the max memcap, we get used data */
-        if (!(THASH_CHECK_MEMCAP(ctx, THASH_DATA_SIZE(ctx)))) {
+        if (!(THASH_CHECK_MEMCAP(ctx, THASH_DATA_SIZE(ctx) + len))) {
             h = THashGetUsed(ctx);
```

Comment:
in this path, where the data size added to the memcap?

Reply:
It was missing indeed. In master too.
I added that in a fixup commit. Please check. The assertion trigger still happens.
Please note that the cases where this is happening, this call path was not followed.

Reply:
In master we don't need it? The ThashData object is not alloc'd, so already part of the memcap